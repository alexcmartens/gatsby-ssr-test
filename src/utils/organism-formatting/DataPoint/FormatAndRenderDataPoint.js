import React from "react"
import formatTheme from "../formatTheme"
import renderPaddingOrFullBleed from "../renderPaddingOrFullBleed"
import ColorSchemeConnector from "../../../components/ColorSchemeConnector/ColorSchemeConnector"
import formatIcons from "../formatIcons"
import { DataPoint } from "terra-one"

const RenderDataPointComponent = ({ points, theme, variant }) => {
  return <DataPoint points={points} theme={theme} variant={variant} />
}

const FormatAndRenderDataPoint = ({
  data,
  disablePadding,
  overrideTheme,
  overrideVariant,
}) => {
  data["theme"] =
    overrideTheme || ColorSchemeConnector(data?.colorScheme)?.theme

  const theme = formatTheme(data)
  const points = data?.points.map(point => {
    return {
      header: point?.header,
      description: point?.description,

      icon: { type: formatIcons(point?.dataPointIcon?.icon) },
    }
  })

  const variant = overrideVariant || formatVariantToMachineName(data?.variant)

  return disablePadding === true ? (
    <RenderDataPointComponent points={points} theme={theme} variant={variant} />
  ) : (
    renderPaddingOrFullBleed(
      <RenderDataPointComponent
        points={points}
        theme={theme}
        variant={variant}
      />,
      true,
      false,
      false,
      false
    )
  )
}

export default FormatAndRenderDataPoint

export const formatIconToMachineName = icon => {
  if (!icon || typeof icon !== "string") {
    return null
  }
  return `${icon.toLowerCase().replace(/ /g, "-")}-dark-48px`
}

export const formatVariantToMachineName = variant => {
  if (!variant || typeof variant !== "string") {
    return "variant_1"
  }

  if (variant.toLowerCase() === "horizontal") {
    return "variant_1"
  } else if (variant?.toLowerCase() === "vertical long align left (3 points)") {
    return "variant_2a"
  } else if (variant?.toLowerCase() === "vertical long centered (3 points)") {
    return "variant_2b"
  } else if (variant?.toLowerCase() === "vertical short (1 point only)") {
    return "variant_3"
  } else {
    return "variant_1"
  }
}
