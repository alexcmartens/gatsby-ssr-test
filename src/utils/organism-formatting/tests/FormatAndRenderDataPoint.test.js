import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderDataPoint, {
  formatVariantToMachineName,
} from "../DataPoint/FormatAndRenderDataPoint"

describe("FormatAndRenderDataPoint", () => {
  it("renders correctly", () => {
    const points = [
      { item: "item1" },
      { item: "item2" },
      { item: "item3" },
      { item: "item4" },
    ]

    const colorScheme = "white"

    const data = { points, variant: "variant_1", colorScheme }

    const tree = renderer
      .create(<FormatAndRenderDataPoint data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe("formatVariantToMachineName", () => {
  it("should return default variant if no argument is passed into method", () => {
    const result = formatVariantToMachineName()
    expect(result).toEqual("variant_1")
  })

  it("should return correct variant when string is passed in", () => {
    const result = formatVariantToMachineName("Horizontal")
    expect(result).toEqual("variant_1")

    const result1 = formatVariantToMachineName(
      "Vertical Long align left (3 points)"
    )
    expect(result1).toEqual("variant_2a")

    const result2 = formatVariantToMachineName(
      "Vertical Long centered (3 points)"
    )
    expect(result2).toEqual("variant_2b")

    const result3 = formatVariantToMachineName("Vertical Short (1 point only)")
    expect(result3).toEqual("variant_3")
  })

  it("should return variant_1 if incorrect string is passed into method", () => {
    const result = formatVariantToMachineName("Variant 1000")
    expect(result).toEqual("variant_1")
  })
})
