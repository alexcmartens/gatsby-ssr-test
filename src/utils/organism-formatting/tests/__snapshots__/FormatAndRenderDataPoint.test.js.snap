// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderDataPoint renders correctly 1`] = `
<section
  className="default-layout   renderdatapointcomponent
      default-layout--no-bottom-margin"
  style={
    Object {
      "backgroundColor": "",
      "marginBottom": "",
    }
  }
>
  <section
    className="default-layout__grid 
          
        "
    style={
      Object {
        "backgroundColor": "",
        "marginBottom": "",
      }
    }
  >
    <div
      className="default-layout__children "
    >
      <div
        points={
          Array [
            Object {
              "description": undefined,
              "header": undefined,
              "icon": Object {
                "type": undefined,
              },
            },
            Object {
              "description": undefined,
              "header": undefined,
              "icon": Object {
                "type": undefined,
              },
            },
            Object {
              "description": undefined,
              "header": undefined,
              "icon": Object {
                "type": undefined,
              },
            },
            Object {
              "description": undefined,
              "header": undefined,
              "icon": Object {
                "type": undefined,
              },
            },
          ]
        }
        theme={
          Object {
            "accordionIconColor": "#0063A3",
            "backgroundColor": "#FFFFFF",
            "color": "#252A2E",
            "ctaOne": "primary--1",
            "ctaTwo": "primary--2",
            "darkTheme": false,
            "iconColor": "#252A2E",
            "theme": "white",
          }
        }
        variant="variant_1"
      />
    </div>
  </section>
</section>
`;
